libr/core/io.c

```diff
@@ -39,10 +41,15 @@ R_API int r_core_setup_debugger (RCore *r, const char *debugbackend, bool attach
 		}
 	}
 	r_core_cmd (r, "sr PC", 0);
-	if (r_config_get_i (r->config, "dbg.status"))
-		r_config_set (r->config, "cmd.prompt", ".dr*;drd;sr PC;pi 1;s-");
-	else
-		r_config_set (r->config, "cmd.prompt", ".dr*");
+
+	/* set the prompt if it's not been set already by the callbacks */
+	prompt = r_config_get (r->config, "cmd.prompt");
+	if (prompt && !strcmp (prompt, "")) {
+		if (r_config_get_i (r->config, "dbg.status"))
+			r_config_set (r->config, "cmd.prompt", ".dr*;drd;sr PC;pi 1;s-");
+		else
+			r_config_set (r->config, "cmd.prompt", ".dr*");
+	}
```

Comment:
I'm really against ternary operator use, especially one this complicated. Further, this is totally irrelevant to the changeset.